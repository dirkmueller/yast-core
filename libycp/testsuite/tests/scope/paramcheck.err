[Interpreter] tests/scope/paramcheck.ycp:15 excessive parameters in function call
[Scanner] tests/scope/paramcheck.ycp:15 Error: Function type mismatch
[Scanner] tests/scope/paramcheck.ycp:15 Error: Expected '**nothing**', seen 'string'.
[Interpreter] tests/scope/paramcheck.ycp:16 excessive parameters in function call
[Scanner] tests/scope/paramcheck.ycp:16 Error: Parameter type mismatch
[Scanner] tests/scope/paramcheck.ycp:16 Error: Expected '**nothing**', seen 'integer'.
[Scanner] tests/scope/paramcheck.ycp:17 Error: Missing 'integer' parameter.
[Scanner] tests/scope/paramcheck.ycp:18 Error: Parameter type mismatch
[Scanner] tests/scope/paramcheck.ycp:18 Error: Expected 'integer', seen 'string'.
Parsed:
----------------------------------------------------------------------
{
    // integer FortyTwo()
    // integer Add()
    // filename: "tests/scope/paramcheck.ycp"
    integer FortyTwo ()
{
    return 42;
}
    integer Add (integer a, integer b)
{
    return (a + b);
}
    return [FortyTwo (), Add (20, 3), YError, YError, YError, YError];
}
----------------------------------------------------------------------
