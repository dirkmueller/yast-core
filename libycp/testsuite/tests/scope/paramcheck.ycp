// I think there's a bug in checking function parameters
// mvidner@suse.cz
{
    define integer FortyTwo () ``{
	return 42;
    }

    define integer Add (integer a, integer b) ``{
	return a + b;
    }

    return [
	FortyTwo (), 		// OK
	Add (20, 3),		// OK
	FortyTwo ("nasty"),	// too many
	Add (1, 2, 3),		// too many
	Add (17),		// too few
	Add (1, "one"),		// mismatch
	];
}
