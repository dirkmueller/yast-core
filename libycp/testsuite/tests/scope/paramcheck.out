([42, 23, *** Type mismatch, *** Type mismatch, *** Missing argument, *** Type mismatch])
